modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -607,12 +607,57 @@ def validate_confinements!(service_information)
                                                  form_object_desc)
           end
 
-          if begin_date_is_not_after_end_date?(approximate_begin_date, approximate_end_date)
+          if begin_date_is_after_end_date?(approximate_begin_date, approximate_end_date)
             raise ::Common::Exceptions::UnprocessableEntity.new(
-              detail: 'Approximate end date must be after approximate begin date.'
+              detail: 'Confinement approximate end date must be after approximate begin date.'
             )
           end
+
+          service_periods = service_information&.dig('servicePeriods')
+          earliest_active_duty_begin_date = find_earliest_active_duty_begin_date(service_periods)
+          # if confinementBeginDate is before earliest activeDutyBeginDate, raise error
+          if begin_date_is_after_end_date?(earliest_active_duty_begin_date['activeDutyBeginDate'],
+                                           approximate_begin_date)
+            raise ::Common::Exceptions::UnprocessableEntity.new(
+              detail: 'Confinement approximate begin date must be after earliest active duty begin date.'
+            )
+          end
+
+          unless confinement_dates_are_within_service_period?(approximate_begin_date, approximate_end_date,
+                                                              service_periods)
+            raise ::Common::Exceptions::UnprocessableEntity.new(
+              detail: 'Confinement dates must be within one of the service period dates.'
+            )
+          end
+        end
+      end
+
+      def confinement_dates_are_within_service_period?(approximate_begin_date, approximate_end_date, service_periods)
+        service_periods.each do |sp|
+          active_duty_begin_date = Date.strptime(sp['activeDutyBeginDate'], '%m-%d-%Y')
+          active_duty_end_date = Date.strptime(sp['activeDutyEndDate'], '%m-%d-%Y')
```

Comment:
Looking at the v2 docs it looks like activeDutyBeginDate and activeDutyEndDate values are not required. So it might be worth it to put a check in for presence before trying to use Date.strtptime on them. I was able to generate an error by removing activeDutyEndDate which then is calling this on nil at this line.

Reply:
Discussed with Jennica; the current 526.json doesn't list activeDutyBeginDate and activeDutyEndDate values as nullable, and a lot of our existing/pending validation (confinements, treatments, BDD program type) would break. Also got an error from EVSS docker when I set activeDutyEndDate to null during testing. Leaving out the check for now.